homeassistant/components/mqtt/config_flow.py

```diff
@@ -338,28 +332,66 @@ async def async_step_options(
         )
 
 
-def try_connection(
+async def async_get_broker_settings(
+    hass: HomeAssistant,
+    fields: OrderedDict[Any, Any],
     yaml_config: ConfigType,
-    broker: str,
-    port: int,
-    username: str | None,
-    password: str | None,
-    protocol: str = "3.1",
+    entry_config: MappingProxyType[str, Any] | None,
+    user_input: ConfigType | None,
```

Comment:
We try to avoid using the type variables for other parameters than its specific parameter. `ConfigType` is meant for the dict imported from the YAML config.